tools/codegen/api/python.py

```diff
@@ -189,26 +236,56 @@ class PythonArgument:
 
     # Compute argument formal for python argument parsing.
     # Needs to be consistent with torch/csrc/utils/python_arg_parser.h.
-    def argument_str(self, *, method: bool = False) -> str:
-        type_str = argument_type_str(self.type)
+    def argument_str(self, *, method: bool = False, pyi: bool = False, deprecated: bool = False) -> str:
```

Comment:
(not in this PR) curious - why not split into argument_str_pyi() if we have 'if pyi' on almost every section :)

Reply:
I actually had a separate `argument_str_pyi()` earlier. I just tried to merge the two to get a better idea of what the differences with pyi were (and whether some differences could be eliminated without byte-for-byte output changes).

That was helpful for the comparison, but I can make it a separate function now that it's pretty clear there are still a bunch of differences

Reply:
@bdhirsh I had an earlier comment about the differences between py->cpp codegen and the pyi codegen: https://github.com/pytorch/pytorch/pull/48754#pullrequestreview-545013186 - it doesn't show as inline comment on the diff view so in case you miss it.
The main question there was that if we are intentionally generating PythonArgParser schema differently from the python interface (and no plan to completely consolidate these two), then we probably should probably keep them separate.
After you removed the 'pyi' bool from the data model, I think the current status seems good - we have the same data models but different set of presenters to generate the two types of signature strings. This single place seems to break the symmetry so asking... :)